app/src/main/java/co/devskills/springbootboilerplate/service/CreditDataAPIClient.java

```diff
@@ -0,0 +1,24 @@
+package co.devskills.springbootboilerplate.service;
+
+import org.springframework.cloud.openfeign.FeignClient;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+
+import co.devskills.springbootboilerplate.constants.ApplicationConstants;
+import co.devskills.springbootboilerplate.models.DebtDetails;
+import co.devskills.springbootboilerplate.models.IncomeDetails;
+import co.devskills.springbootboilerplate.models.PersonalDetails;
+
+@FeignClient(name = "credit-data-api", url = ApplicationConstants.BASE_URL)
+public interface CreditDataAPIClient {
```

Comment:
modified from the spring rest template to feign client for better abstraction of the rest endpoint calls. does not need any unit tests etc